src/main.rs

```diff
@@ -78,31 +62,8 @@ fn main() {
 
     let player = Player::new(&session);
 
-    SpircManager {
-        session: &session,
-        player: &player,
-
-        username: username.clone(),
-        state_update_id: 0,
-        seq_nr: 0,
-
-        name: name.clone(),
-        ident: session.config.device_id.clone(),
-        device_type: 5,
-        can_play: true,
-
-        repeat: false,
-        shuffle: false,
-        volume: 0x8000,
-
-        is_active: false,
-        became_active_at: 0,
-
-        last_command_ident: String::new(),
-        last_command_msgid: 0,
-
-        track: None
-    }.run();
+    let mut spirc_manager = SpircManager::new(&session, &player, username, name);
```

Comment:
Had to add `mut` to be able to call `run()` on it after, let me know if I'm doing something wrong here.